services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    /**
+     * Returns a signed URL with a canned policy that grants universal access to
+     * private content until a given date.
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-canned-policy.html"
+     * >Creating a signed URL using a canned policy</a>.
+     *
+     * @param request
+     *            A CloudFrontSignerRequest configured with the following values:
+     *            resourceUrl, privateKey, keyPairId, expirationDate
+     * @return A signed URL that will permit access to a specific distribution
+     *         and S3 object.
+     */
+    public static SignedUrl getSignedUrlWithCannedPolicy(CloudFrontSignerRequest request) {
+        try {
+            String resourceUrl = request.resourceUrl();
+            String cannedPolicy = SigningUtils.buildCannedPolicy(resourceUrl, request.expirationDate());
+            byte[] signatureBytes = SigningUtils.signWithSha1Rsa(cannedPolicy.getBytes(UTF_8), request.privateKey());
+            String urlSafeSignature = SigningUtils.makeBytesUrlSafe(signatureBytes);
+            String protocol = resourceUrl.substring(0, resourceUrl.indexOf("://"));
+            String domain = resourceUrl.substring(resourceUrl.indexOf("://") + 3, resourceUrl.indexOf(CLOUDFRONT_NET) + 14);
+            String encodedPath = resourceUrl.substring(resourceUrl.indexOf(CLOUDFRONT_NET) + 15)
+                   + (request.resourceUrl().indexOf('?') >= 0 ? "&" : "?")
+                   + "Expires=" + request.expirationDate().getEpochSecond()
+                   + "&Signature=" + urlSafeSignature
+                   + "&Key-Pair-Id=" + request.keyPairId();
+            return DefaultSignedUrl.builder().protocol(protocol).domain(domain).encodedPath(encodedPath).build();
+        } catch (InvalidKeyException e) {
+            throw SdkClientException.create("Could not sign url", e);
+        }
+    }
+
+    /**
+     * Returns a signed URL that provides tailored access to private content
+     * based on an access time window and an ip range. The custom policy itself
+     * is included as part of the signed URL (For a signed URL with canned
+     * policy, there is no policy included in the signed URL).
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-custom-policy.html"
+     * >Creating a signed URL using a custom policy</a>.
+     *
+     * @param request
+     *            A CloudFrontSignerRequest configured with the following values:
+     *            resourceUrl, privateKey, keyPairId, expirationDate, activeDate, ipRange
+     * @return A signed URL that will permit access to distribution and S3
+     *         objects as specified in the policy document.
+     */
+    public static SignedUrl getSignedUrlWithCustomPolicy(CloudFrontSignerRequest request) {
```

Comment:
Should these take different request objects, and return different response objects? Using the same requests is potentially confusing since the type system doesn't give the caller any hints about what fields are required. Using the same response is okay for now, but if these methods ever need to return something different, we'll be trapped.

Reply:
The same request object makes more sense to me since a request for custom policy is just an extension of a request for canned policy, with the option to add active date and IP range (you could add either, both, or none).

Agree with having different response objects in case of future modifications, will split into two classes for URLs for canned and custom policies

Reply:
More than anything, a different request object makes more sense to me. There can be a common base class, but you don't want people passing an activeDate and ipRange to `getSignedUrlWithCannedPolicy`

Reply:
I hear you, keeping one single response object, creating separate request objects for canned/custom